bin.src/create_agn_db.py

```diff
@@ -208,30 +208,100 @@ def get_m_i(abs_mag_i, redshift):
         m_i_grid[i_z] = ss.calcMag(bp_dict['i'])
         mag_norm_grid[i_z] = ss.calcMag(imsimband)
 
-    interpolated_m_i = np.interp(redshift, z_grid, m_i_grid)
-    interpolated_mag_norm = np.interp(redshift, z_grid, mag_norm_grid)
-    mag_norm = interpolated_mag_norm + (obs_mag_i - interpolated_m_i)
-
+    htmid_level = 6
     with sqlite3.connect(out_file_name) as connection:
         cursor = connection.cursor()
         cursor.execute('''CREATE TABLE agn_params
-                          (galaxy_id, magNorm real, varParamStr text)''')
+                          (galaxy_id int, htmid_%d int, magNorm real, varParamStr text)''' % htmid_level)
 
         connection.commit()
 
-        seed_arr = rng.randint(1,high=10000000, size=len(tau))
-
-        vals = ((int(ii), mm, '{"m": "applyAgn", '
-                      + '"p": {"seed": %d, "agn_tau": %.3e, "agn_sfu": %.3e, ' % (ss, tt, sfu)
-                      + '"agn_sfg": %.3e, "agn_sfr": %.3e, "agn_sfi": %.3e, ' % (sfg, sfr, sfi)
-                      + '"agn_sfz": %.3e, "agn_sfy": %.3e}}' % (sfz, sfy))
-                 for ii, mm, ss, tt, sfu, sfg, sfr, sfi, sfz, sfy in
-                 zip(galaxy_id, mag_norm, seed_arr, tau,
-                     sf_dict['u'], sf_dict['g'], sf_dict['r'], sf_dict['i'],
-                     sf_dict['z'], sf_dict['y']))
-
-        cursor.executemany('INSERT INTO agn_params VALUES(?, ?, ?)', vals)
+        chunk_size = 100000
+        full_size = len(redshift_full)
+        print('starting iteration')
+        t_start = time.time()
+        ct_simulated = 0
+        for i_start in range(0, full_size, chunk_size):
+            i_end = i_start + chunk_size
+            duration = (time.time()-t_start)/3600.0
+            per = duration/(1+i_start)
+            predicted = full_size*per
+            print("    %d through %d -- took %.2e hrs; predict %.2e" %
+            (i_start,i_end,duration,predicted))
+
+            galaxy_id = galaxy_id_full[i_start:i_end]
+            ra = ra_full[i_start:i_end]
+            dec = dec_full[i_start:i_end]
+            ct_simulated += len(galaxy_id)
+            redshift = redshift_full[i_start:i_end]
+            log_edd_ratio = log_edd_ratio_full[i_start:i_end]
+            bhm = bhm_full[i_start:i_end]
```

Comment:
not important either --- but can consider define `selection = slice(i_start, i_end)`

Reply:
Cool.  I didn't know about `slice`.  Thanks